src/container/container_iv.c

```diff
@@ -1092,6 +1084,40 @@ cont_iv_prop_fetch(struct ds_iv_ns *ns, uuid_t cont_hdl_uuid,
 	return rc;
 }
 
+/*
+ * exported APIs
+ */
+int
+ds_cont_fetch_snaps(struct ds_iv_ns *ns, uuid_t cont_uuid,
+		    uint64_t **snapshots, int *snap_count)
+{
+	D_ASSERT(dss_get_module_info()->dmi_xs_id == 0);
+	return cont_iv_snapshots_fetch(ns, cont_uuid, snapshots, snap_count);
+}
+
+int
+ds_cont_revoke_snaps(struct ds_iv_ns *ns, uuid_t cont_uuid,
+		     unsigned int shortcut, unsigned int sync_mode)
+{
+	D_ASSERT(dss_get_module_info()->dmi_xs_id == 0);
```

Comment:
probably these assert(xs_id == 0) should be moved to iv_op_internal(), i.e. only stream 0 is allowed to do iv ops.